modules/api/src/main/java/org/apache/ignite/table/partition/HashPartition.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.table.partition;
+
+import java.io.Serializable;
+import org.apache.ignite.table.criteria.CriteriaVisitor;
+import org.apache.ignite.table.criteria.Partition;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Hash partition representation.
+ */
+public class HashPartition implements Partition, Serializable {
+    private static final long serialVersionUID = 1717320056615864614L;
+
+    private final int partitionId;
```

Comment:
Do we actually need a class to wrap an integer? The API would be much simpler with `int` / `Integer` without all the generics for partition type.

Do we expect other types of partitions in the API?

Reply:
What is a range partition, do you have more details?

Reply:
Ok, and we'll be using partitions identified by `int` anyway.

Reply:
Range partition is identfied by (lower, upper) bounds.

Reply:
Ok, we should keep the classes then.